.github/workflows/api-compatibility.yml

```diff
@@ -31,9 +31,9 @@ jobs:
           path: ${{ github.head_ref }}
 
       - name: Setup Go
-        uses: actions/setup-go@f111f3307d8850f501ac008e886eec1fd1932a34 # v5.3.0
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
         with:
-          go-version: 1.23.9
+          go-version: 1.24.3
```

Comment:
blocked by go1.24

Reply:
Why?

Reply:
It seems we had similar conversations before https://github.com/open-telemetry/opentelemetry-collector/pull/12655 & https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38726 

Not sure if the above still holds? cc @atoulme @codeboten

Reply:
Added an issue to discuss the tooling changes https://github.com/open-telemetry/opentelemetry-collector/issues/13200, @atoulme @songy23 @bogdandrutu ptal